k8s/jenkins/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,250 @@
 actions:
-- name: dummy test
+- name: Jenkins service login
   bashTest:
     script: |-
-      exit 0
+      SERVICE="${APP_INSTANCE_NAME}-jenkins-ui"
+      IP=''
+      PORT=''
+      PASSWORD=''
+      FAILED=0
+
+      function wait_for_service_ip {
+        IP=$(kubectl get service "${SERVICE}" \
+          --namespace "${NAMESPACE}" \
+          -ojsonpath="{.spec.clusterIP}")
+        if [[ $? -ne 0 ]]; then
+          IP=""
+        fi
+      }
+
+      function wait_for_service_port {
+        PORT=$(kubectl get service "${SERVICE}" \
+          --namespace "${NAMESPACE}" \
+          -ojsonpath="{.spec.ports[0].port}")
+        if [[ $? -ne 0 ]]; then
+          PORT=""
+        fi
+      }
+
+      function get_jenkins_pod {
+        POD=$(kubectl get pod -oname \
+          --namespace "${NAMESPACE}" | \
+          sed -n "/\\/${APP_INSTANCE_NAME}-jenkins-deployment/s.pods\\?/..p")
+        if [[ $? -ne 0 ]]; then
+          POD=""
+        fi
+      }
+
+      function get_jenkins_admin_password {
+        PASSWORD=$(kubectl exec "${POD}" \
```

Comment:
Could you explain why do we need loop here? It seems to me like the `kubectl exec` command is possibile at once, after a `POD` is working,

Reply:
To keep consistency with the rest of the code.